lib/build-cloud.rb

```diff
@@ -220,16 +223,31 @@ def recursive_interpolate_config(h)
         when String
 
             while h =~ /%\{(.+?)\}/
```

Comment:
Also, I'd need to be able to pull out the entire string for substitution later on in the `gsub`, and wouldn't this just mean equivalent effort to recreate that string later on vs. splitting it out in the code?